swarm/network/kademlia.go

```diff
@@ -367,12 +371,15 @@ func (k *Kademlia) EachBin(base []byte, pof pot.Pof, o int, eachBinFunc func(con
 	kadDepth := depthForPot(k.conns, k.MinProxBinSize, k.base)
 
 	k.conns.EachBin(base, pof, o, func(po, size int, f func(func(val pot.Val, i int) bool) bool) bool {
-		if startPo > 0 && endPo != k.MaxProxDisplay {
-			startPo = endPo + 1
-		}
+		//if the peer's bin is smaller than the kademlia depth,
+		//only the peer's bin should be subscribed
```

Comment:
I stopped short here in review because I realized it's in progress.

Just wanted to mention that "subscribe" is out of context in kademlia code. Please let's not add comments in terms of consumer components.